### PR TITLE
[CHORE] Update Storybook 7.5.3

### PR DESCRIPTION
**Pull request by ahna92**
### Describe your changes
* Update typescript 5.2.2
* Update React 18.2.0
* Update storybook 
* Update awesome fonts 
* Add iOS 17 OS support
* Add iOS 15 pro + max
* Add android 13 OS
* Add pixel  7
* Node 18 #78
### UI changes
image

### Testing
* example android material ui
* example ios material ui
* example flutter
* example controls
* example cross platform 

<details>
  <summary>📦 Published PR as canary version: <code>2.2.9-canary.79.843.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/native-android-material-deep-link-example@2.2.9-canary.79.843.0
  npm install @storybook/native-controls-example@2.2.9-canary.79.843.0
  npm install @storybook/native-cross-platform-example@2.2.9-canary.79.843.0
  npm install @storybook/native-flutter-example@2.2.9-canary.79.843.0
  npm install @storybook/native-ios-example-deep-link@2.2.9-canary.79.843.0
  npm install @storybook/native-addon@2.2.9-canary.79.843.0
  npm install @storybook/native-controllers@2.2.9-canary.79.843.0
  npm install @storybook/deep-link-logger@2.2.9-canary.79.843.0
  npm install @storybook/native-dev-middleware@2.2.9-canary.79.843.0
  npm install @storybook/native-devices@2.2.9-canary.79.843.0
  npm install @storybook/native-components@2.2.9-canary.79.843.0
  npm install @storybook/native@2.2.9-canary.79.843.0
  npm install @storybook/native-types@2.2.9-canary.79.843.0
  # or 
  yarn add @storybook/native-android-material-deep-link-example@2.2.9-canary.79.843.0
  yarn add @storybook/native-controls-example@2.2.9-canary.79.843.0
  yarn add @storybook/native-cross-platform-example@2.2.9-canary.79.843.0
  yarn add @storybook/native-flutter-example@2.2.9-canary.79.843.0
  yarn add @storybook/native-ios-example-deep-link@2.2.9-canary.79.843.0
  yarn add @storybook/native-addon@2.2.9-canary.79.843.0
  yarn add @storybook/native-controllers@2.2.9-canary.79.843.0
  yarn add @storybook/deep-link-logger@2.2.9-canary.79.843.0
  yarn add @storybook/native-dev-middleware@2.2.9-canary.79.843.0
  yarn add @storybook/native-devices@2.2.9-canary.79.843.0
  yarn add @storybook/native-components@2.2.9-canary.79.843.0
  yarn add @storybook/native@2.2.9-canary.79.843.0
  yarn add @storybook/native-types@2.2.9-canary.79.843.0
  ```
</details>

### CODE REVIEW

**Comment by ahna92** [comment]
Hi @sumwatshade 
Can you help me checking what is the issue with release  step

**Comment by ahna92** [comment]
hi @amalik2 
can you help with creating a canary for this build , would like to test it on my projects

**Comment by chriskurzeja** [comment]
We'd love to see these updates get merged as we're having great success with storybook native in our [design system](https://engineering.countingup.com/design-system/). Thanks @ahna92 for raising this.

**Comment by amalik2** [comment]
Hi @ahna92 @chriskurzeja sorry for the delay, a canary version is available for consumption now (it's listed at the bottom of the PR body).

I'll merge this as a major release if the canary version looks good on your end

**Comment by ahna92** [comment]
hi @amalik2 
I faced compatibility issues when tried to use `7.5.2` so i upgraded to `7.5.3` , let's see if this will help 

But need your help with another canary please , 

issue i faced : 

```
TypeError: Cannot read properties of undefined (reading 'getAddonState')
    at useSharedState (http://localhost:9006/vendors-node_modules_storybook_addon-docs_dist_preview_mjs-node_modules_storybook_addon-essen-6844e3.iframe.bundle.js:1167:39945)
    at useAddonState (http://localhost:9006/vendors-node_modules_storybook_addon-docs_dist_preview_mjs-node_modules_storybook_addon-essen-6844e3.iframe.bundle.js:1167:42176)
    at DeviceDecorator (http://localhost:9006/vendors-node_modules_storybook_addon-docs_dist_preview_mjs-node_modules_storybook_addon-essen-6844e3.iframe.bundle.js:1237:53)
    at hookified (http://localhost:9006/sb-preview/runtime.js:7:18973)
    at http://localhost:9006/sb-preview/runtime.js:41:1451
    at http://localhost:9006/sb-preview/runtime.js:41:1912
    at withOutline (http://localhost:9006/vendors-node_modules_storybook_addon-docs_dist_preview_mjs-node_modules_storybook_addon-essen-6844e3.iframe.bundle.js:976:669)
    at hookified (http://localhost:9006/sb-preview/runtime.js:7:18973)
    at http://localhost:9006/sb-preview/runtime.js:41:1451
    at http://localhost:9006/sb-preview/runtime.js:41:1912
```

image

**Comment by amalik2** [comment]
@ahna92 Updated, the new version is `2.2.9-canary.79.827.0`

**Comment by ahna92** [comment]
hi @chriskurzeja 
I've tested the migration on one of my storybook projects and it's working (still have another 2 to check) 
Can you help testing it too , check the [migration steps](https://github.com/ahna92/native/blob/update_storybook/MIGRATION.md)

The migration steps might need some tuning (use yarn add instead of copy/paste)

@amalik2 
issue above resolved, also noticed the [argtypes structure](https://storybook.js.org/docs/react/api/arg-types#argtypes) was changes so had to update the loadash templates

**Comment by chriskurzeja** [comment]
@ahna92 I see you found the issue with `deviceDecorator`. I hadn't had chance to dig into it so nice one for spotting it.

**Comment by chriskurzeja** [comment]
@ahna92 @amalik2 I've tested `2.2.9-canary.79.829.0` with our design system and it works - the issue we'd had was related to the deviceDecorator but that's now patched. This is great!

File: MIGRATION.md [hunk 1]
```diff
@@ -1,5 +1,90 @@
 # Migration Guide
 
+## Version `6.*` to Version `7.*`
```
Comment (amalik2) [inline MIGRATION.md]: Could you change this to `Version 2 to Version 3` so that its consistent with the existing migration guide?

**Review by amalik2 — CHANGES_REQUESTED** [review]
LGTM but one comment on the migration guide

File: MIGRATION.md (hunk 1 above)
Comment (ahna92) [reply MIGRATION.md]: done

**Review by amalik2 — APPROVED** [review]

**Comment by sumwatshade** [comment]
:rocket: PR was released in [`v3.0.0`](https://github.com/storybookjs/native/releases/tag/v3.0.0) :rocket:

**Pull request merged**